chain/chain/src/resharding/manager.rs

```diff
@@ -103,6 +105,63 @@ impl ReshardingManager {
         Ok(())
     }
 
+    /// Returns the list of ShardUIds that need to be loaded into memory at
+    /// client startup. This list consists of all ShardUIds in the current shard
+    /// layout that are planned to be resharded within this client.
+    ///
+    /// e.g. In the following resharding tree shards 0 and 1 would be returned.
+    ///
+    ///  0      1       2
+    ///  |     / \      |
+    ///  0    3   4     2
+    ///  |\   |   |     |
+    ///  5 6  3   4     2
+    ///  | |  |   |\    |
+    ///  5 6  3   7 8   2
+    ///
+    /// Please note that shard 4 is not returned even though it is split later
+    /// on. That is because it is a child of another parent and it should
+    /// already be loaded into memory after the first resharding.
+    pub fn get_extra_load_mem_tries_for_shards(
+        &self,
+        head_protocol_version: ProtocolVersion,
+        client_protocol_version: ProtocolVersion,
+    ) -> Result<HashSet<ShardUId>, Error> {
+        let mut shard_layouts = vec![];
+        let mut protocol_version = head_protocol_version + 1;
+        while protocol_version <= client_protocol_version {
+            let shard_layout =
+                self.epoch_manager.get_shard_layout_from_protocol_version(protocol_version)?;
+
+            let last_shard_layout = shard_layouts.last();
+            if last_shard_layout == None || last_shard_layout != Some(&shard_layout) {
+                shard_layouts.push(shard_layout);
+            }
+
+            protocol_version += 1;
+        }
+
+        let mut result = HashSet::new();
+        let head_shard_layout =
+            self.epoch_manager.get_shard_layout_from_protocol_version(head_protocol_version)?;
+        for shard_uid in head_shard_layout.shard_uids() {
+            let shard_id = shard_uid.shard_id();
+            for shard_layout in &shard_layouts {
+                let children = shard_layout.get_children_shards_uids(shard_id);
+                let Some(children) = children else {
+                    break;
+                };
+                if children.len() > 1 {
```

Comment:
sanity check, it can only be 0 or 2, right?

Reply:
Today it can only be 1 or 2. 1 means no spit, the shard stays as is and 2 mean split into two children. In the future we'll have plenty of fun with merging though ;)